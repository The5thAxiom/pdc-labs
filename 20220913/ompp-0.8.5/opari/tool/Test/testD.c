void foo (bool b) {
  int i;

  #pragma pomp inst begin(foo)
  i = 4;
  if (b) {
    #pragma pomp inst altend(foo)
    return;
  }

  #pragma omp parallel
  {
    #pragma omp inst begin(phase1)

    #pragma omp barrier

    #pragma omp inst end(phase1)
  }

  if (b) {
    #pragma pomp inst altend(foo)
    return;
  }

  #pragma pomp inst end(foo)
}
