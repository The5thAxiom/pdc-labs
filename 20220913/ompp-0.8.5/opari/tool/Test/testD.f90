subroutine foo(b)
  logical b
  integer i

  !$POMP INST BEGIN(foo)
  i = 4
  if ( b ) then
    !$POMP INST ALTEND(foo)
    return
  end if

  !$OMP PARALLEL

    !$OMP INST BEGIN(phase1)

    !$OMP BARRIER

    !$OMP INST END(phase1)

  !$OMP END PARALLEL

  if ( b ) then
    !$POMP INST ALTEND(foo)
    return
  end if

  !$POMP INST END(foo)
end subroutine foo
