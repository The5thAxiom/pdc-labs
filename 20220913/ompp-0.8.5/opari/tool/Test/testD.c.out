#ifdef _POMP
#  undef _POMP
#endif
#define _POMP 200110

#include "testD.c.opari.inc"
#line 1 "testD.c"
void foo (bool b) {
  int i;

POMP_Begin(&omp_rd_40);
#line 5 "testD.c"
  i = 4;
  if (b) {
POMP_End(&omp_rd_40);
#line 8 "testD.c"
    return;
  }

POMP_Parallel_fork(&omp_rd_41);
#line 11 "testD.c"
  #pragma omp parallel POMP_DLIST_00041
{ POMP_Parallel_begin(&omp_rd_41);
#line 12 "testD.c"
  {
POMP_Begin(&omp_rd_42);
#line 14 "testD.c"

POMP_Barrier_enter(&omp_rd_43);
#line 15 "testD.c"
    #pragma omp barrier
POMP_Barrier_exit(&omp_rd_43);
#line 16 "testD.c"

POMP_End(&omp_rd_42);
#line 18 "testD.c"
  }
POMP_Barrier_enter(&omp_rd_41);
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_41);
POMP_Parallel_end(&omp_rd_41); }
POMP_Parallel_join(&omp_rd_41);
#line 19 "testD.c"

  if (b) {
POMP_End(&omp_rd_40);
#line 22 "testD.c"
    return;
  }

POMP_End(&omp_rd_40);
#line 26 "testD.c"
}
